lib/src/simple_op_heads_store.rs

```diff
@@ -83,24 +84,41 @@ impl OpHeadsStore for SimpleOpHeadsStore {
         Self::name()
     }
 
-    fn update_op_heads(&self, old_ids: &[OperationId], new_id: &OperationId) {
+    fn update_op_heads(
+        &self,
+        old_ids: &[OperationId],
+        new_id: &OperationId,
+    ) -> Result<(), OpHeadsStoreError> {
         assert!(!old_ids.contains(new_id));
         self.add_op_head(new_id);
         for old_id in old_ids {
             self.remove_op_head(old_id);
         }
+        Ok(())
     }
 
-    fn get_op_heads(&self) -> Vec<OperationId> {
+    fn get_op_heads(&self) -> Result<Vec<OperationId>, OpHeadsStoreError> {
         let mut op_heads = vec![];
-        for op_head_entry in std::fs::read_dir(&self.dir).unwrap() {
-            let op_head_file_name = op_head_entry.unwrap().file_name();
-            let op_head_file_name = op_head_file_name.to_str().unwrap();
+        for op_head_entry in
+            std::fs::read_dir(&self.dir).map_err(|err| OpHeadsStoreError::Read(err.into()))?
+        {
+            let op_head_file_name = op_head_entry
+                .map_err(|err| OpHeadsStoreError::Read(err.into()))?
+                .file_name();
+            let op_head_file_name = op_head_file_name.to_str().ok_or_else(|| {
+                OpHeadsStoreError::Read(
+                    format!(
+                        "Non-utf8 in op head file name: {}",
+                        op_head_file_name.to_string_lossy()
```

Comment:
nit: maybe okay to use `{op_head_file_name:?}` here? it's known to contain unprintable characters.

Reply:
Yes, that's better. Done.